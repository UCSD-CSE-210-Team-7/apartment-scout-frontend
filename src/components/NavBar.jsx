// Import required dependencies, components and assets.
import React, { useContext } from "react";
import { Link, useLocation } from "react-router-dom";
import Auth from "../utils/auth";
import "../styles/NavBar.css";

/**
 * NavigationBar component displays the name of the website "aptcout". It contains three buttons on
 * the top right namely the home, chat and account buttons. All these buttons are linked to various
 * web pages. The home button will always bring the user back to the home page where they enter the
 * zip codes. the chat button will open the chat box where all the messages are saved. The account button
 * has 3 more options namely profile, tours and logout.
 * The navigation bar is consistent to all pages and always present on every page.
 * @returns {JSX.Element} The JSX element for the HomePageComponent.
 */

const NavigationBar = () => {
  const { logout } = useContext(Auth);
  const { pathname } = useLocation();

  const NavLink = ({ selected, href, match = [], children, style = {} }) => {
    const colorScheme =
      selected || pathname === href || match.includes(pathname)
        ? { color: "#023E8A", background: "#ADE8F4" }
        : { color: "#FFFFFF" };
    return (
      <Link to={href}>
        <span
          style={{
            cursor: "pointer",
            padding: "0 1vw",
            ...colorScheme,
            ...style,
          }}
        >
          {children}
        </span>
      </Link>
    );
  };

  if (pathname === "/") {
    return;
  }

  return (
    <nav
      style={{
        height: "10vh",
        alignItems: "center",
        padding: "1vw 3vw",
        background: "#023E8A",
        display: "flex",
        justifyContent: "space-between",
      }}
    >
      {/* The links to all the various webpages are defined here on clicking the different functionalities. */}
      <Link to="/home">
        <div>
          <span style={{ fontSize: "3.5em", color: "#ADE8F4" }}>apt</span>
          <span style={{ fontSize: "3.5em", color: "#FFFFFF" }}>scout</span>
        </div>
      </Link>
      <div style={{ display: "flex", alignItems: "center" }}>
        <NavLink href="/home">home</NavLink>
        <NavLink href="/chat">chat</NavLink>
        <div className="d">
          <NavLink href="/profile" match={["/profile", "/tours"]}>
            account
          </NavLink>
          <div className="dc">
            <NavLink href="/profile">profile</NavLink>
            <NavLink href="/tourhome">tours</NavLink>
            <span
              style={{ cursor: "pointer", color: "#FFFFFF" }}
              onClick={logout}
            >
              logout
            </span>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default NavigationBar;
