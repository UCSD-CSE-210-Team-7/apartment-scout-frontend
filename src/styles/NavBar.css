nav span {
  font-size: 3em;
}

nav a {
  color: inherit;
  text-decoration: none;
}

button {
  background-color: #ffb511;
  color: black;
  border-radius: 6px;
}

a:hover {
  text-decoration: none;
}

.d {
  position: relative;
}

.dc {
  display: none;
  position: absolute;
  right: 0;
  background: rgb(2, 62, 138);
  border: 1px solid white;
}

.dc span {
  display: block;
  text-align: center;
  padding: 0 0vw;
}

.d:hover .dc {
  display: flex;
  flex-direction: column;
}

/*
.dropdown {
    display: inline-block;
    width: 100%;
}

.dropdown-content {
    display: none;
    position: absolute;
    right: 0;
    min-width: 160px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
    z-index: 3;
}

.dropdown-content button {
    color: black;
    padding: 12px 16px;
    text-decoration: none;
    display: block;
    width: 100%;
    background-color: #ffb511;
    min-width: 160px;
    box-shadow: 0px 8px 16px 0px rgba(1,1,1,0.2);
    z-index: 3;
    border-radius: 0px;
}

.dropdown-content button:hover {
    background-color: #aaa;
    box-shadow: 0px 8px 16px 0px rgba(1,1,1,0.2);
    z-index: 3;
}
.nav-bar{
   backgroundColor: "rgb(23,40,74)"
}
*/
