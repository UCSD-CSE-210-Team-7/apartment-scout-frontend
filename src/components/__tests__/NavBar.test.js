import {screen, act} from '@testing-library/react'
import { render } from '../../test_setup'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

import Auth from '../../utils/auth';
import NavBar from '../NavBar';

test("render", () => {
  const { asFragment } = render( <NavBar /> );
  expect(asFragment()).toMatchSnapshot()
});

test("click buttons", async () => {
  const { asFragment } = render( <NavBar /> );
  expect(screen.getByRole('link', {name: 'apt scout'})).toHaveAttribute('href', '/home')

  expect(screen.getByRole('link', {name: 'home'})).toHaveAttribute('href', '/home')

  expect(screen.getByRole('link', {name: 'chat'})).toHaveAttribute('href', '/chat')

  expect(screen.getByRole('link', {name: 'profile'})).toHaveAttribute('href', '/profile')

  expect(screen.getByRole('link', {name: 'tours'})).toHaveAttribute('href', '/tours')
});
