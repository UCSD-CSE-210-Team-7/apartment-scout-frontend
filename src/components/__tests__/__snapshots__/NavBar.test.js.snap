// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<DocumentFragment>
  <nav
    style="height: 10vh; align-items: center; padding: 1vw 3vw; background: rgb(2, 62, 138); display: flex; justify-content: space-between;"
  >
    <a
      href="/home"
    >
      <div>
        <span
          style="font-size: 3.5em; color: rgb(173, 232, 244);"
        >
          apt
        </span>
        <span
          style="font-size: 3.5em; color: rgb(255, 255, 255);"
        >
          scout
        </span>
      </div>
    </a>
    <div
      style="display: flex; align-items: center;"
    >
      <a
        href="/home"
      >
        <span
          style="cursor: pointer; padding: 0px 1vw; color: rgb(2, 62, 138); background: rgb(173, 232, 244);"
        >
          home
        </span>
      </a>
      <a
        href="/chat"
      >
        <span
          style="cursor: pointer; padding: 0px 1vw; color: rgb(255, 255, 255);"
        >
          chat
        </span>
      </a>
      <div
        class="d"
      >
        <a
          href="/profile"
        >
          <span
            style="cursor: pointer; padding: 0px 1vw; color: rgb(255, 255, 255);"
          >
            account
          </span>
        </a>
        <div
          class="dc"
        >
          <a
            href="/profile"
          >
            <span
              style="cursor: pointer; padding: 0px 1vw; color: rgb(255, 255, 255);"
            >
              profile
            </span>
          </a>
          <a
            href="/tours"
          >
            <span
              style="cursor: pointer; padding: 0px 1vw; color: rgb(255, 255, 255);"
            >
              tours
            </span>
          </a>
          <span
            style="cursor: pointer; color: rgb(255, 255, 255);"
          >
            logout
          </span>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
